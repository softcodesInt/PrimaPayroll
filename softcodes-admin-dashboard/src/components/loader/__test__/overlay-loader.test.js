/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

/* -------------------------- Internal Dependencies ------------------------- */
import OverlayLoader from '../overlay';

describe('Overlay Loader Component', () => {
	afterEach(cleanup);

	it('Overlay Loader Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<OverlayLoader loading />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Overlay Loader Renders Correctly With Attributes', () => {
		const { getByTestId } = render(
			<OverlayLoader loadingText="Loading" loading />
		);
		expect(getByTestId('overlay-text')).toHaveTextContent('Loading');
	});

	it('Overlay Loader matches snapshot', () => {
		const tree = renderer
			.create(<OverlayLoader loadingText="Loading" loading />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
