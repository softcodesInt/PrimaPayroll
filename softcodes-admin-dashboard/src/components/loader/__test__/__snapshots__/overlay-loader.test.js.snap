// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Loader Component Overlay Loader matches snapshot 1`] = `
Array [
  " ",
  <div
    className="sc-bdnylx GCHgv"
  >
    <span>
      <svg>
        loading.svg
      </svg>
      <h3
        data-testid="overlay-text"
      >
        Loading
      </h3>
    </span>
  </div>,
]
`;
